src/gen9_avc_encoder.c

```diff
@@ -5399,9 +5435,12 @@ gen9_avc_encode_check_parameter(VADriverContextP ctx,
     /*slice check,all the slices use the same slice height except the last slice*/
     avc_state->arbitrary_num_mbs_in_slice = 0;
     for (i = 0; i < avc_state->slice_num; i++) {
-        assert(avc_state->slice_param[i]->num_macroblocks % generic_state->frame_width_in_mbs == 0);
-        avc_state->slice_height = avc_state->slice_param[i]->num_macroblocks / generic_state->frame_width_in_mbs;
-        /*add it later for muli slices map*/
+        if (avc_state->slice_param[i]->num_macroblocks % generic_state->frame_width_in_mbs > 0) {
+            avc_state->arbitrary_num_mbs_in_slice = 1;
+            avc_state->slice_height = 1;
```

Comment:
In this case, the kernel does not care this value of slice_height, here i just put 1 as the default value